R-package/configure.ac

```diff
@@ -0,0 +1,167 @@
+then
+    ac_pkg_r35=yes
+    LGB_CPPFLAGS+=" -DR_VER_ABOVE_35"
+fi
+AC_MSG_RESULT([${ac_pkg_r35}])
+
+# Compile for GPU?
+AC_ARG_ENABLE(
+    [gpu],
+    AS_HELP_STRING(
+        [--enable-gpu],
+        [Pass this flag to install the R package for usage with GPU]
+    ),
+    [
+        LGB_CPPFLAGS+=" -DUSE_GPU"
+    ]
+)
+
+###############
+# MM_PREFETCH #
+###############
+
+AC_MSG_CHECKING([whether MM_PREFETCH works])
+ac_mmprefetch=no
+AC_LANG_CONFTEST(
+    [
+        AC_LANG_PROGRAM(
+            [[
+                #include <xmmintrin.h>
+            ]],
+            [[
+                int main() {
+                    int a = 0;
+                    _mm_prefetch(&a, _MM_HINT_NTA);
+                    return 0;
+                }
+            ]]
+        )
+    ]
+)
+${CC} -o conftest conftest.c 2>/dev/null && ./conftest && ac_mmprefetch=yes
+AC_MSG_RESULT([${ac_mmprefetch}])
+if test "${ac_mmprefetch}" = yes; then
+    LGB_CPPFLAGS+=" -DMM_PREFETCH=1"
+fi
+
+############
+# MM_ALLOC #
+############
+
+AC_MSG_CHECKING([whether MM_MALLOC works])
+ac_mm_malloc=no
+AC_LANG_CONFTEST(
+    [
+        AC_LANG_PROGRAM(
+            [[
+                #include <mm_malloc.h>
+            ]],
+            [[
+                int main() {
+                    char *a = (char*)_mm_malloc(8, 16);
+                    _mm_free(a);
+                    return 0;
+                }
+            ]]
+        )
+    ]
+)
+${CC} -o conftest conftest.c 2>/dev/null && ./conftest && ac_mm_malloc=yes
+AC_MSG_RESULT([${ac_mm_malloc}])
+if test "${ac_mm_malloc}" = yes; then
+    LGB_CPPFLAGS+=" -DMM_MALLOC=1"
+fi
+
+##########
+# OpenMP #
+##########
+
+OPENMP_CXXFLAGS=""
+
+if test `uname -s` = "Linux"
+then
+    OPENMP_CXXFLAGS="\$(SHLIB_OPENMP_CXXFLAGS)"
+fi
+
+if test `uname -s` = "Darwin"
+then
+    OPENMP_CXXFLAGS='-Xclang -fopenmp'
+    OPENMP_LIB='/usr/local/lib/libomp.dylib'
+    ac_pkg_openmp=no
+    AC_MSG_CHECKING([whether OpenMP will work in a package])
+    AC_LANG_CONFTEST(
+        [
+            AC_LANG_PROGRAM(
+                [[
+                    #include <omp.h>
+                ]],
+                [[
+                    return (omp_get_max_threads() <= 1);
+                ]]
+            )
+        ]
+    )
+    ${CC} -o conftest conftest.c ${OPENMP_LIB} ${OPENMP_CXXFLAGS} 2>/dev/null && ./conftest && ac_pkg_openmp=yes
+    AC_MSG_RESULT([${ac_pkg_openmp}])
+    if test "${ac_pkg_openmp}" = no; then
+        OPENMP_CXXFLAGS=''
+        OPENMP_LIB=''
+        echo '***********************************************************************************************'
+        echo ' OpenMP is unavailable on this Mac OSX system. LightGBM code may be slower as a result'
+        echo ' To use all CPU cores for training jobs, you should install OpenMP by running'
+        echo ''
+        echo '     brew install libomp'
+        echo '***********************************************************************************************'
+    fi
+fi
```

Comment:
I took this directly from our friends at `xgboost`: https://github.com/dmlc/xgboost/blob/master/R-package/configure.ac#L29-L47